slsa-framework/slsa-github-generator
# ci: Add formatting pre-submit check

# Summary

Adds a formatting pre-submit check that runs `make format` and checks for file differences. The results of running `make format` are not included in this PR for readability. So the pre-submit will depend on #3655 and probably some more formatting before the check will pass.

Added a `make shfmt` formatting check for bash scripts.

`autogen` (for adding license headers) is now vendored in the `third_party` directory for convenience sake.

`markdown-toc` and `autogen` are now run by `make format`. The formatting pre-submit now runs these as well so they don't have separate pre-submit jobs. We will need to adjust the required checks on branch protection to remove "markdown-toc" and "autogen" jobs, and also add the "formatting" job.

`CONTRIBUTING.md` was updated to include some more information about coding style and conventions.

## Testing Process

1. Run `make format`
2. Verify the formatting results

## Checklist

- [x] Review the contributing [guidelines](https://github.com/slsa-framework/slsa-github-generator/blob/main/CONTRIBUTING.md)
- [x] Add a reference to related issues in the PR description.
- [x] Update documentation if applicable.
- [x] Add unit tests if applicable.
- [x] Add changes to the [CHANGELOG](https://github.com/slsa-framework/slsa-github-generator/blob/main/CHANGELOG.md) if applicable.
